### DISCUSSION TITLE
Add a option to filter out folders when updating models/LoRas

### DISCUSSION THREAD

**Discussion by CesarCruzUtec** [BlafKing/sd-civitai-browser-plus#142] [Ideas]
I have downloaded some LoRas from other sources like HugginFace in a separate folder inside the LoRas folder, and therefore there is no link to its counterpart in Civitai. In my case, this new folder has over 150 loras, and when trying to look for updates for the civitai ones, it takes a long time searching for the other ones. So a suggestion might be to add an option to filter out folders  or specific files, for the checkpoints maybe, and avoid the long time to wait.

![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/67798431/43f4e113-75e6-4741-a7ac-18e13a3eac5e)
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/67798431/b8a8bcb0-7237-43e8-8534-34a3c4abacee)
![image](https://github.com/BlafKing/sd-civitai-browser-plus/assets/67798431/ded4b6c8-c5bc-4286-81f6-0387d142ff33)

**Comment by BlafKing** [comment]
Thanks for the suggestion!

The wait time is long because it fetches available updates by using the files SHA256 hash, if a file doesn't have a saved SHA256 hash yet, then it generates one and saves it to a corresponding .json file, the larger the file, the longer it will take to generate one.

It only needs to do this once for each file that wasn't downloaded with the extension, so after you have processed all the files once, then it will be much faster the next time!

I'll add a toggle in a future update that you can use to disable generating a hash for external downloads/files, I personally recommend generating a hash for each file though, since a lot of models are often on both HuggingFace and CivitAI :)

**Comment by BlafKing** [comment]
[v3.2.0](https://github.com/BlafKing/sd-civitai-browser-plus/commit/d21758863a0605f5139b1f5f49af0a4e00880b02) now includes a toggle which can be used to disable creating new hashes for any files that don't have any hash saved.
This will cause those files to be skipped in any of the update models functions!
